lib/core/BlockchainClock.ts

```diff
@@ -63,10 +62,16 @@ export default class BlockchainClock {
         EventEmitter.emit(EventCode.SidetreeBlockchainTimeChanged, { time: latestBlockchainTime.time });
       } catch (e) {
         Logger.error(`Error occured while updating blockchain time, investigate and fix: ${e}`);
-      } finally {
-        if (this.continuePulling) {
-          setTimeout(async () => this.startPeriodicPullLatestBlockchainTime(), this.blockchainTimePullIntervalInSeconds * 1000);
-        }
+      }
+    }
+
+    private async cacheTime () {
+      try {
+        const newApproximateTime = (await this.serviceStateStore.get()).approximateTime;
```

Comment:
Both `cacheTime()` and `pullRealBlockchainTime` need this, so maybe just do it once? To me the main logic is more of an if-else thing:

```typescript
if (this.enableRealBlockchainTimePull) {
  // Update in-memory time if needed by pulling from bitcoin.
} else {
 // Update in-memory time from DB.
}